test/test_models_transform.py

```diff
@@ -19,3 +22,32 @@ def test_yolo_transform():
     # Test annotations after transformation
     torch.testing.assert_close(annotations[0]["boxes"], annotations_copy[0]["boxes"], rtol=0, atol=0)
     torch.testing.assert_close(annotations[1]["boxes"], annotations_copy[1]["boxes"], rtol=0, atol=0)
+
+
+@pytest.mark.parametrize("img_h", [300, 500, 720, 800, 1080, 1280])
+@pytest.mark.parametrize("img_w", [300, 500, 720, 800, 1080, 1280])
+@pytest.mark.parametrize("auto", [True])
+@pytest.mark.parametrize("stride", [32, 64])
+def test_letterbox(img_h, img_w, auto, stride):
+
+    from yolort.models.transform import _resize_image_and_masks
+    from yolort.v5 import letterbox
+
+    new_shape = (640, 640)  # height, width
+
+    img_tensor = torch.randint(0, 255, (3, img_h, img_w))
+    img_numpy = img_tensor.permute(1, 2, 0).numpy().astype("uint8")
+
+    yolo_transform = YOLOTransform(new_shape[0], new_shape[1], size_divisible=stride, auto_rectangle=auto)
+
+    im3 = img_tensor / 255
+    im3, _ = _resize_image_and_masks(im3.float(), new_shape)
+    out1 = yolo_transform.batch_images([im3])
+
+    out2 = letterbox(img_numpy, new_shape=new_shape, auto=auto, stride=stride)
+
+    aug1 = out1[0].numpy()
+    aug2 = out2[0].astype(np.float32)  # uint8 to float32
+    aug2 = np.transpose(aug2 / 255.0, [2, 0, 1])
+    assert aug1.shape == aug2.shape
+    np.testing.assert_allclose(aug1, aug2, rtol=1e-4, atol=1e-2)
```

Comment:
PyTorch's [interpolate](https://pytorch.org/docs/stable/generated/torch.nn.functional.interpolate.html#torch-nn-functional-interpolate) operator now only aligns with OpenCV for the float type, and the `letterbox` implemented by yolov5 operates on integers, so we set a lower precision here. Check https://github.com/pytorch/pytorch/issues/5580#issuecomment-907051670 for more details.